services/TS29222_CAPIF_Logging_API_Invocation_API/api_invocation_logs/core/invocationlogs.py

```diff
@@ -0,0 +1,58 @@
+import sys
+import os
+import pymongo
+import secrets
+from flask import current_app, Flask, Response
+import json
+
+from ..db.db import MongoDatabse
+from ..encoder import JSONEncoder
+from ..models.problem_details import ProblemDetails
+
+
+class LoggingInvocationOperations:
+
+    def __init__(self):
+        self.db = MongoDatabse()
```

Comment:
The connection to the database is not made from the class but from an abstract class and we make the class that performs the operations on the database inherit from it (Review code)